openwisp_users/admin.py

```diff
@@ -139,10 +178,22 @@ def has_change_permission(self, request, obj=None):
         return super(UserAdmin, self).has_change_permission(request, obj)
 
     def get_queryset(self, request):
-        qs = super(UserAdmin, self).get_queryset(request)
-        # hide superusers from operators (they can't edit their details)
+        """
+        if operator is logged in - show only users
+        from same organization and hide superusers
+        if superuser is logged in - show all users
+        """
         if not request.user.is_superuser:
-            qs = qs.filter(is_superuser=False)
+            org_ids = OrganizationUser.objects.only("organization_id") \
+                                              .values_list("organization_id") \
+                                              .filter(user=request.user,
+                                                      is_admin=True)
+            user_ids = OrganizationUser.objects.only("user_id") \
+                                               .values_list("user_id") \
+                                               .filter(organization__in=org_ids)
+            qs = User.objects.filter(pk__in=user_ids, is_superuser=False)
```

Comment:
querying users by a list of IDs will be a problem in instances which use the freeradius moduile in which we expect organizations to have many users (potentially hundred of thousands). This code won't scale there because the full list of users for an organization will have to be fetched and built before being able to even query the DB and with a very large set of users this process can become slow, even cause timeouts of requests or halting the application server.

That's why last week I pushed this commit: https://github.com/openwisp/openwisp-users/commit/55ad1b3d9a199c33fa8e66517b51f0e363456561 which I think handled the problem well: for each organization the user is member of, a queryset is built to fetch users of that org except the requesting user. That queryset is not evaluated until executed so it will perform well. The code I wrote is more complex but sometimes that's a cost we can't avoid to sustain in order to write software which scales decently.

So what prompted you to replace that code with this code here?

Reply:
PS: look at the current file you're editing, if single quotes `'` are used to delimit strings, please follow the same style. Let's not mix styles.

Reply:
> So what prompted you to replace that code with this code here?

The problem with that code was that when a user has multiple common organisations with current operator, multiple records (for each organisation) are displayed on list and selecting any instance of such user throws an error.
However, i've been working on this problem and it looks like it can simply be solved by doing
`add_qs = org_user.organization.users.all().distinct()` -- i'll test it further and revert back to changes in that commit if everything works out, if not i'll discuss the problem here!